packages/devtools_app/lib/src/shared/editable_list.dart

```diff
@@ -0,0 +1,272 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import 'common_widgets.dart';
+import 'theme.dart';
+import 'utils.dart';
+
+class EditableList extends StatefulWidget {
+  const EditableList({
+    required this.entries,
+    required this.textFieldLabel,
+    this.isRefreshing,
+    this.onEntryAdded,
+    this.onEntryRemoved,
+    this.onRefresh,
+  });
+
+  final ValueListenable<List<String>> entries;
+  final ValueListenable<bool>? isRefreshing;
+  final Function(String)? onEntryAdded;
+  final Function(String)? onEntryRemoved;
```

Comment:
on second thought, should we allow the EditableList to modify the ListValueNotifier directly, and then we wouldn't need to pass these callbacks through? Then on add and delete the editable list code could just call entried.add and entries.remove

Reply:
I'm not sure that would be a better approach. 

My first concern is that making the widget the source of truth, and then having to interpret whether something was added or removed after the list is updated, would make the contents of the list unreliable.
This would be further complicated by the fact that letting the list in the widget be a source of truth, means that we have 2 sources of truth that we need to keep in sync. The directories on the service and the directories in the widget. We would need to have extra logic to handle the cases when they get out of sync, or ways for us to know they get out of sync.

Reply:
> making the widget the source of truth

The widget wouldn't be the source of truth, the ListValueNotifier would be the source of truth. The only change here is that the widget has write access to the ListValueNotifier instead of having to go through the controller as a middleman.

Reply:
we chatted about this on Video chat and came a compromise where default values will be added on onAdded and onRemoved.
That way this common widget can be more easily dropped into place and interacted with, when there aren't complex async interactions for add and remove